### DISCUSSION TITLE
Extremely Long Push Time (20-30 Minutes) Resulting in Failure

### DISCUSSION THREAD

**Discussion by MatinT-SA** [community/community#138627] [Repositories]
### Select Topic Area

Question

### Body

Hello,

I’m encountering an issue with pushing my commits to the remote repository on GitHub. Initially, I suspected that large files in my local repository were the cause. I removed all video files and decided to store them on YouTube, embedding them on my website instead. However, this didn’t resolve the problem.

I also followed recommendations to increase the http.postBuffer, but that didn’t help either. Additionally, I explored tools like git-filter-repo and BFG to rewrite the commit history, but none of these solutions worked for me. Now, I’ve been unable to push my recent changes for almost 24 hours.

I have also run the garbage collection command:
`git gc --prune=now --aggressive`

Despite this, the issue persists.
I would greatly appreciate any advice or solutions from professionals or anyone who has faced a similar problem.

For reference, here is a screenshot showing where the push process gets stuck:
![github push issue](https://github.com/user-attachments/assets/8151a8d4-1b52-45b9-ae42-d6d3b4e5959a)

**Comment by lqduy** [comment]
Try using a VPN

**Reply by MatinT-SA** [reply]
> Try using a VPN

Well, the interesting part is that, at first I tried to push the commits using VPN and after I read it somewhere that probably VPNs or proxies are causing prolongation, I tried it out without VPNs connection too.
None of them have helped so far.

**Comment by MatinT-SA** [comment]
https://dev.to/dev_neil_a/how-to-clear-a-stuck-github-push-h57

Finally this article helped me to solve this weird issue.
It's a strange behavior from GitHub but if you try to reset to the last commit and then see the untrakced files through `git status` command and try to add them five by five as the author has stated here and then follow the push instructions, you're good to go and also you can get back to adding all your files at once by `git add .` command after that without any issues.
